File: 250504/겹치지 않는 사각형의 넓이/area-of-non-overlapping-rectangle.java
import java.util.Scanner;

public class Main {

    public static final int MAX_N = 2001;
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        int offset = 1000;
        boolean[][] area = new boolean[MAX_N][MAX_N];
        int ax1 = sc.nextInt() + offset;
        int ay1 = sc.nextInt() + offset;
        int ax2 = sc.nextInt() + offset;
        int ay2 = sc.nextInt() + offset;
        int bx1 = sc.nextInt() + offset;
        int by1 = sc.nextInt() + offset;
        int bx2 = sc.nextInt() + offset;
        int by2 = sc.nextInt() + offset;
        int mx1 = sc.nextInt() + offset;
        int my1 = sc.nextInt() + offset;
        int mx2 = sc.nextInt() + offset;
        int my2 = sc.nextInt() + offset;
        for(int i = ax1; i < ax2; i++) {
            for (int j = ay1; j < ay2; j++) {
                area[i][j] = true;
            }
        }
        for(int i = bx1; i < bx2; i++) {
            for (int j = by1; j < by2; j++) {
                area[i][j] = true;
            }
        }
        for(int i = mx1; i < mx2; i++) {
            for (int j = my1; j < my2; j++) {
                area[i][j] = false;
            }
        }
        int uncoveredArea = 0;
        for(int i = 0; i < MAX_N; i++) {
            for (int j = 0; j < MAX_N; j++) {
                if (area[i][j]) {
                    uncoveredArea++;
                }
            }
        }
        System.out.println(uncoveredArea);
        sc.close();
    }
}

File: 250504/겹치지 않는 사각형의 넓이/area-of-non-overlapping-rectangle.cpp
#include <iostream>

#define MAX_N 2001
#define OFFSET 1000 // 좌표를 음수로 만들지 않기 위해

using namespace std;

int x1, y1, x2, y2; // 직사각형의 좌표

bool area[MAX_N][MAX_N];

int main()
{
    ios::sync_with_stdio(0);
    cin.tie(NULL);
    cout.tie(NULL);

    for(int i = 0; i < 3; i++) {
        cin >> x1 >> y1 >> x2 >> y2;
        if(i < 2) { // 첫 번째와 두 번째 직사각형은 겹치는 부분을 true로 설정
            for(int j = x1 + OFFSET; j < x2 + OFFSET; j++) {
                for(int k = y1 + OFFSET; k < y2 + OFFSET; k++) {
                    area[j][k] = true; // 직사각형의 영역을 true로 설정
                }
            }
        } else { // 세 번째 직사각형은 겹치는 부분을 false로 설정
            for(int j = x1 + OFFSET; j < x2 + OFFSET; j++) {
                for(int k = y1 + OFFSET; k < y2 + OFFSET; k++) {
                    area[j][k] = false;
                }
            }
        }
    }
    int count = 0;
    for(int i = 0; i < MAX_N; i++) {
        for(int j = 0; j < MAX_N; j++) {
            if(area[i][j]) { // true인 부분의 개수를 세기
                count++;
            }
        }
    }
    cout << count << "\n"; // 결과 출력
    return 0;
}